benchbuild/environments/entrypoints/cli.py

```diff
@@ -62,21 +62,21 @@ def main(self, *projects: str) -> int:
             print("No projects selected.")
             return -2
 
-        create_base_images(wanted_projects)
-        create_project_images(wanted_projects)
+        create_base_images(wanted_experiments, wanted_projects)
+        create_project_images(wanted_experiments, wanted_projects)
         create_experiment_images(wanted_experiments, wanted_projects)
         run_experiment_images(wanted_experiments, wanted_projects)
 
         return 0
 
 
 def enumerate_projects(
-    projects: ProjectIndex
+    experiments: ExperimentIndex, projects: ProjectIndex
 ) -> tp.Generator[project.Project, None, None]:
-    for prj_class in projects.values():
-        for variant in source.product(*prj_class.SOURCE):
-            context = source.context(*variant)
-            yield prj_class(context)
+    for exp_class in experiments.values():
+        for prj_class in projects.values():
+            for context in exp_class.sample(prj_class):
+                yield prj_class(context)
```

Comment:
Yes, this is normal. I'd like to move the older interfaces to use instances instead of classes too. Seems a bit cleaner and now projects/experiments support this.